In [32]:
import pandas as pd
# from src.utils import calculate_campign_metrics

In [33]:
df = pd.read_csv(r'C:\Users\karthikeya\health_tech_analytics\results\signup_counts.csv')

In [34]:
df

campaign_id        campaign_category         campaign_type         cost  \
0       CAM001    Tailored Health Plans       Covid Awareness  1335.566784   
1       CAM002     Preventive Care News  Customer Testimonial  1624.442464   
2       CAM003     Preventive Care News    Policy Information  1944.226008   
3       CAM004  Compare Health Coverage  Customer Testimonial  1354.937186   
4       CAM005    Golden Years Security    Offer Announcement  1519.167326   
5       CAM006  Compare Health Coverage     Product Promotion  1254.286130   
6       CAM007           #HealthyLiving    Policy Information  1173.585931   
7       CAM008     Preventive Care News    Policy Information   223.085599   
8       CAM009    Tailored Health Plans      Health Awareness   394.641083   
9       CAM010         Affordable Plans       Covid Awareness   820.580377   
10      CAM011    Golden Years Security    Offer Announcement  1334.056787   
11      CAM012         #CoverageMatters     Product Promotion   244.008996   
12      CAM013        #InsureYourHealth  Customer Testimonial  1134.316809   
13      CAM014     Summer Wellness Tips    Policy Information   125.046745   
14      CAM015     Summer Wellness Tips      Health Awareness  1555.577561   
15      CAM016  Compare Health Coverage       Covid Awareness  1367.173425   
16      CAM017    Tailored Health Plans    Offer Announcement  1089.283035   
17      CAM018           Health For All      Health Awareness  1185.171031   
18      CAM019        #InsureYourHealth       Covid Awareness  1872.242670   
19      CAM020           #HealthyLiving           Health Tips  1822.723467   
20      CAM021     Family Coverage Plan  Customer Testimonial  1606.227668   
21      CAM022        #InsureYourHealth     Product Promotion   738.830067   
22      CAM023           #HealthyLiving    Offer Announcement   303.769053   
23      CAM024          Benefit Updates     Product Promotion   710.720045   
24      CAM025     Family Coverage Plan  Customer Testimonial  1528.728281   
25      CAM026         #CoverageMatters       Covid Awareness  1260.901406   
26      CAM027          Benefit Updates    Policy Information  1440.586227   
27      CAM028         Affordable Plans    Policy Information   300.126002   
28      CAM029    Tailored Health Plans           Health Tips   401.049088   
29      CAM030           #HealthyLiving    Policy Information  1362.207577   
30      CAM031         #CoverageMatters     Product Promotion   806.421044   
31      CAM032     Family Coverage Plan     Product Promotion   801.456917   
32      CAM033           Health For All     Product Promotion   846.052987   
33      CAM034  Compare Health Coverage    Policy Information  1612.285770   
34      CAM035  Compare Health Coverage  Customer Testimonial  1428.202582   
35      CAM036    Golden Years Security       Covid Awareness   301.467866   
36      CAM037  Compare Health Coverage           Health Tips   754.278902   
37      CAM038     Summer Wellness Tips    Policy Information  1380.281030   
38      CAM039    Tailored Health Plans       Covid Awareness  1954.645480   
39      CAM040     Summer Wellness Tips       Covid Awareness  1833.281138   
40      CAM041    Golden Years Security       Covid Awareness   571.719463   
41      CAM042        #InsureYourHealth    Offer Announcement   225.462663   
42      CAM043        #InsureYourHealth     Product Promotion  1189.200451   
43      CAM044  Compare Health Coverage       Covid Awareness   871.875628   
44      CAM045         Affordable Plans       Covid Awareness   448.441110   
45      CAM046     Preventive Care News       Covid Awareness  1338.129684   
46      CAM047           #HealthyLiving           Health Tips   404.666776   
47      CAM048     Preventive Care News     Product Promotion   767.814707   
48      CAM049           #HealthyLiving    Offer Announcement  1609.678559   
49      CAM050  Compare Health Coverage     Product Promotion  1400.168346   
50      CAM051           

In [35]:
def calculate_campign_metrics(df : pd.DataFrame)->pd.DataFrame:
    df['cost_per_impression'] = df['cost']/df['impressions']
    df['cost_per_click'] = df['cost']/df['clicks']
    df['customer_acqisition_cost'] = df['cost']/df['num_signups']
    df['click_through_rate'] = (df["clicks"]/df["impressions"])*100
    df['signup_rate'] = (df['num_signups']/ df['impressions'])*100

    return df

In [36]:
df.select_dtypes(exclude="object").columns

Index(['cost', 'impressions', 'clicks', 'days_run', 'num_signups'], dtype='object')

In [37]:
columns = ['cost', 'impressions', 'clicks', 'num_signups']

In [38]:
df.sort_values(by="num_signups",ascending=False).head(10)
df.drop(columns=['campaign_id_1'],inplace=True)

## we have highest signup rate on E-mail platform with Health Awareness being the type of campaign, followed by social media with policy information and SEO with product info

In [39]:
df.groupby(['platform','campaign_category'])[['cost','impressions' ,'clicks','num_signups']].sum().sort_values(by="num_signups",ascending=False)

cost  impressions    clicks  \
platform     campaign_category                                              
Social Media #HealthyLiving            6676.631363      1372448  132067.5   
             #CoverageMatters          2311.331446       712100   74240.0   
Email        Health For All            1185.171031        88220   43453.0   
SEO          Compare Health Coverage  10043.207969       664710   93349.5   
             Tailored Health Plans     5175.185470      1398859   92570.0   
Email        Preventive Care News      6256.810157      1116968  136663.0   
Social Media #InsureYourHealth         6572.881885       970244   74540.0   
TV           Health For All            3161.795270        82339       0.0   
             Family Coverage Plan       801.456917       452680       0.0   
Email        Summer Wellness Tips      7085.680129       573010  103667.5   
SEO          Affordable Plans          1569.147489       327507   41593.5   
Email        Benefit Updates           2151.306272       244280   54145.0   
SEO          Family Coverage Plan      1606.227668       449340       0.0   
Social Media Family Coverage Plan      1528.728281       204520       0.0   
SEO          Golden Years Security     2853.224113       348510    5979.0   
TV           Golden Years Security     1211.462688        74420       0.0   

                                      num_signups  
platform     campaign_category                     
Social Media #HealthyLiving                3727.0  
             #CoverageMatters              3536.0  
Email        Health For All                3279.0  
SEO          Compare Health Coverage       2820.0  
             Tailored Health Plans         1107.0  
Email        Preventive Care News           643.0  
Social Media #InsureYourHealth              316.0  
TV           Health For All                 266.0  
             Family Coverage Plan           228.0  
Email        Summer Wellness Tips           163.0  
SEO          Affordable Plans                63.0  
Email        Benefit Updates                 45.0  
SEO          Family Coverage Plan            42.0  
Social Media Family Coverage Plan            31.0  
SEO          Golden Years Security           23.0  
TV           Golden Years Security            0.0

In [40]:
import plotly.express as px
# import plotly.graph_objects as go

In [41]:
df_sunburst = df.groupby(['platform','campaign_type'])[columns].sum().sort_values(by="num_signups",ascending=False).reset_index()  # Replace with your actual data

# Choose the value column to represent size (e.g., cost_sum)
fig = px.sunburst(
    df_sunburst,
    path=["platform", "campaign_type"],  # Hierarchy levels
    values="num_signups",                     # Column for size (replace with your metric)
    title="campaign signup rates 360"
    # color="cost",                   # Optional: Column for color-coding metrics (replace with actual column name)
    # color_discrete_sequence=px.colors.qualitative.Plotly  # Optional: Color palette
)
fig.show()
fig.write_image("C:/Users/karthikeya/health_tech_analytics/plots/campaign signup rates 360 categorised by campaign type.png")
fig.write_html("C:/Users/karthikeya/health_tech_analytics/plots/campaign signup rates 360 categorised by campaign type.html")

In [42]:
fig  = px.pie(
    df,
    names="platform",
    values="num_signups",
    title="signup rates by platform category"
)
fig.show()
# fig.write_image("C:/Users/karthikeya/health_tech_analytics/signup rates by platform category.png")
# fig.write_html("C:/Users/karthikeya/health_tech_analytics/signup rates by platform category.html")

In [43]:
df_platform=df.groupby(['platform'])[columns].sum().sort_values(by="num_signups",ascending=False)
df_platform= calculate_campign_metrics(df_platform)
df_platform

cost  impressions    clicks  num_signups  \
platform                                                         
Social Media  17089.572975      3259312  280847.5       7610.0   
Email         16678.967588      2022478  337928.5       4130.0   
SEO           21246.992710      3188926  233492.0       4055.0   
TV             5174.714875       609439       0.0        494.0   

              cost_per_impression  cost_per_click  customer_acqisition_cost  \
platform                                                                      
Social Media             0.005243        0.060850                  2.245673   
Email                    0.008247        0.049356                  4.038491   
SEO                      0.006663        0.090997                  5.239702   
TV                       0.008491             inf                 10.475131   

              click_through_rate  signup_rate  
platform                                       
Social Media            8.616772     0.233485  
Email                  16.708637     0.204205  
SEO                     7.321964     0.127159  
TV                      0.000000     0.081058

In [44]:
df_campaign_type=df.groupby(['campaign_type'])[columns].sum().sort_values(by="num_signups",ascending=False)
df_campaign_type=calculate_campign_metrics(df_campaign_type)
df_campaign_type

cost  impressions    clicks  num_signups  \
campaign_type                                                            
Product Promotion     10965.805894      1562824  105050.5       5581.0   
Health Awareness       4520.521931       820960  124169.0       3395.0   
Policy Information    10814.963725      1816321  265360.0       2987.0   
Customer Testimonial   8676.854990      1511676  109730.0       2229.0   
Covid Awareness       13976.025031      1514805  125856.5       1064.0   
Health Tips            5154.659153       719101   50398.0        972.0   
Offer Announcement     6081.417424      1134468   71704.0         61.0   

                      cost_per_impression  cost_per_click  \
campaign_type                                               
Product Promotion                0.007017        0.104386   
Health Awareness                 0.005506        0.036406   
Policy Information               0.005954        0.040756   
Customer Testimonial             0.005740        0.079075   
Covid Awareness                  0.009226        0.111047   
Health Tips                      0.007168        0.102279   
Offer Announcement               0.005361        0.084813   

                      customer_acqisition_cost  click_through_rate  \
campaign_type                                                        
Product Promotion                     1.964846            6.721838   
Health Awareness                      1.331523           15.124854   
Policy Information                    3.620678           14.609752   
Customer Testimonial                  3.892712            7.258831   
Covid Awareness                      13.135362            8.308429   
Health Tips                           5.303147            7.008473   
Offer Announcement                   99.695368            6.320496   

                      signup_rate  
campaign_type                      
Product Promotion        0.357110  
Health Awareness         0.413540  
Policy Information       0.164453  
Customer Testimonial     0.147452  
Covid Awareness          0.070240  
Health Tips              0.135169  
Offer Announcement       0.005377

In [45]:
df_campaign_category=df.groupby(['campaign_category'])[columns].sum().sort_values(by="num_signups",ascending=False)
df_campaign_category = calculate_campign_metrics(df_campaign_category)
df_campaign_category

cost  impressions    clicks  num_signups  \
campaign_category                                                           
#HealthyLiving            6676.631363      1372448  132067.5       3727.0   
Health For All            4346.966301       170559   43453.0       3545.0   
#CoverageMatters          2311.331446       712100   74240.0       3536.0   
Compare Health Coverage  10043.207969       664710   93349.5       2820.0   
Tailored Health Plans     5175.185470      1398859   92570.0       1107.0   
Preventive Care News      6256.810157      1116968  136663.0        643.0   
#InsureYourHealth         6572.881885       970244   74540.0        316.0   
Family Coverage Plan      3936.412866      1106540       0.0        301.0   
Summer Wellness Tips      7085.680129       573010  103667.5        163.0   
Affordable Plans          1569.147489       327507   41593.5         63.0   
Benefit Updates           2151.306272       244280   54145.0         45.0   
Golden Years Security     4064.686801       422930    5979.0         23.0   

                         cost_per_impression  cost_per_click  \
campaign_category                                              
#HealthyLiving                      0.004865        0.050555   
Health For All                      0.025487        0.100038   
#CoverageMatters                    0.003246        0.031133   
Compare Health Coverage             0.015109        0.107587   
Tailored Health Plans               0.003700        0.055906   
Preventive Care News                0.005602        0.045783   
#InsureYourHealth                   0.006774        0.088179   
Family Coverage Plan                0.003557             inf   
Summer Wellness Tips                0.012366        0.068350   
Affordable Plans                    0.004791        0.037726   
Benefit Updates                     0.008807        0.039732   
Golden Years Security               0.009611        0.679827   

                         customer_acqisition_cost  click_through_rate  \
campaign_category                                                       
#HealthyLiving                           1.791422            9.622769   
Health For All                           1.226225           25.476814   
#CoverageMatters                         0.653657           10.425502   
Compare Health Coverage                  3.561421           14.043643   
Tailored Health Plans                    4.674964            6.617536   
Preventive Care News                     9.730653           12.235176   
#InsureYourHealth                       20.800259            7.682604   
Family Coverage Plan                    13.077784            0.000000   
Summer Wellness Tips                    43.470430           18.091744   
Affordable Plans                        24.907103           12.700034   
Benefit Updates                         47.806806           22.165138   
Golden Years Security                  176.725513            1.413709   

                         signup_rate  
campaign_category                     
#HealthyLiving              0.271559  
Health For All              2.078460  
#CoverageMatters            0.496559  
Compare Health Coverage     0.424245  
Tailored Health Plans       0.079136  
Preventive Care News        0.057567  
#InsureYourHealth           0.032569  
Family Coverage Plan        0.027202  
Summer Wellness Tips        0.028446  
Affordable Plans            0.019236  
Benefit Updates             0.018421  
Golden Years Security       0.005438

In [46]:
df_campaign_type_cat=df.groupby(['campaign_type','campaign_category'])[columns].sum().sort_values(by="num_signups",ascending=False)
df_campaign_type_cat=calculate_campign_metrics(df_campaign_type_cat)
df_campaign_type_cat

cost  impressions  \
campaign_type        campaign_category                                   
Product Promotion    #CoverageMatters         1050.430040       446990   
Health Awareness     Health For All           2247.380479       117850   
Policy Information   #HealthyLiving           2535.793508       593220   
Customer Testimonial Compare Health Coverage  2783.139768       140426   
Product Promotion    Compare Health Coverage  2654.454476       116890   
Health Tips          #HealthyLiving           2227.390243       490821   
Covid Awareness      Tailored Health Plans    3290.212264       330069   
Customer Testimonial Preventive Care News     1624.442464       428100   
Product Promotion    Health For All            846.052987        32272   
                     #InsureYourHealth        1928.030518       386223   
                     Family Coverage Plan      801.456917       452680   
Policy Information   Summer Wellness Tips     1505.327775       315285   
Health Awareness     Tailored Health Plans     394.641083       480790   
Customer Testimonial Family Coverage Plan     3134.955949       653860   
Policy Information   Preventive Care News     2167.311607       596980   
Covid Awareness      Affordable Plans         1269.021487       242657   
                     #InsureYourHealth        1872.242670       215500   
Product Promotion    Benefit Updates           710.720045        88390   
Offer Announcement   Tailored Health Plans    1089.283035       451840   
Customer Testimonial #InsureYourHealth        1134.316809       289290   
Offer Announcement   Golden Years Security    2853.224113       348510   
Policy Information   Benefit Updates          1440.586227       155890   
                     Affordable Plans          300.126002        84850   
Offer Announcement   #HealthyLiving           1913.447612       288407   
Health Awareness     Summer Wellness Tips     1878.500369       222320   
Covid Awareness      #CoverageMatters         1260.901406       265110   
Health Tips          Tailored Health Plans     401.049088       136160   
Covid Awareness      Compare Health Coverage  2239.049053       341765   
                     Summer Wellness Tips     1833.281138        19761   
                     Preventive Care News     1338.129684        39003   
                     Golden Years Security     873.187329        60940   
Health Tips          Preventive Care News      359.111694        42630   
Offer Announcement   #InsureYourHealth         225.462663        45711   
Health Tips          #InsureYourHealth        1412.829225        33520   
                     Compare Health Coverage   754.278902        15970   
Policy Information   Compare Health Coverage  1612.285770        49659   
                     Health For All           1253.532835        20437   
Product Promotion    Golden Years Security     338.275359        13480   
                     Preventive Care News      767.814707        10255   
                     Summer Wellness Tips     1868.570846        15644   

                                               clicks  num_signups  \
campaign_type        campaign_category                               
Product Promotion    #CoverageMatters         35655.0       3535.0   
Health Awareness     Health For All           43453.0       3279.0   
Policy Information   #HealthyLiving           58195.0       2745.0   
Customer Testimonial Compare Health Coverage  44160.0       1546.0   
Product Promotion    Compare Health Coverage  25243.5       1273.0   
Health Tips          #HealthyLiving           32857.0        971.0   
Covid Awareness      Tailored Health Plans    22565.0        965.0   
Customer Testimonial Preventive Care News     47330.0        586.0   
Product Promotion    Health For All               0.0        266.0   
                     #InsureYourHealth        29440.0        246.0   
                     Family Coverage Plan         0.0        228.0   
Policy Information   Summer Wellne

In [47]:
df_plat_camp=df.groupby(['platform','campaign_type'])[df.select_dtypes(exclude="object").columns].sum().sort_values(by="num_signups",ascending=False)
df_plat_camp = calculate_campign_metrics(df_plat_camp)
df_plat_camp

cost  impressions    clicks  \
platform     campaign_type                                              
Social Media Product Promotion     2978.460558       833213   65095.0   
Email        Health Awareness      3063.671400       310540   89959.0   
Social Media Policy Information    2535.793508       593220   58195.0   
SEO          Customer Testimonial  4389.367436       589766   44160.0   
             Product Promotion     2654.454476       116890   25243.5   
             Covid Awareness       6798.282804       914491   54274.5   
Social Media Health Tips           3640.219468       524341   33907.5   
Email        Customer Testimonial  1624.442464       428100   47330.0   
TV           Product Promotion     1985.785262       498432       0.0   
Email        Policy Information    5113.225610      1068155  175084.0   
SEO          Health Awareness       394.641083       480790   34210.0   
Social Media Customer Testimonial  2663.045090       493810   18240.0   
SEO          Offer Announcement    3942.507148       800350   29349.0   
Social Media Covid Awareness       3133.144076       480610   63055.0   
Email        Product Promotion     3347.105598       114289   14712.0   
SEO          Policy Information    1912.411772       134509   32081.0   
Social Media Offer Announcement    2138.910275       334118   42355.0   
SEO          Health Tips           1155.327990       152130   14174.0   
Email        Covid Awareness       3171.410822        58764    8527.0   
             Health Tips            359.111694        42630    2316.5   
TV           Covid Awareness        873.187329        60940       0.0   
             Health Awareness      1062.209448        29630       0.0   
             Policy Information    1253.532835        20437       0.0   

                                   days_run  num_signups  cost_per_impression  \
platform     campaign_type                                                      
Social Media Product Promotion           88       3781.0             0.003575   
Email        Health Awareness            56       3281.0             0.009866   
Social Media Policy Information          53       2745.0             0.004275   
SEO          Customer Testimonial        78       1588.0             0.007443   
             Product Promotion           59       1273.0             0.022709   
             Covid Awareness            134       1017.0             0.007434   
Social Media Health Tips                 58        971.0             0.006942   
Email        Customer Testimonial        48        586.0             0.003795   
TV           Product Promotion           66        494.0             0.003984   
Email        Policy Information         139        230.0             0.004787   
SEO          Health Awareness            38        114.0             0.000821   
Social Media Customer Testimonial        48         55.0             0.005393   
SEO          Offer Announcement          94         50.0             0.004926   
Social Media Covid Awareness             53         47.0             0.006519   
Email        Product Promotion           44         33.0             0.029286   
SEO          Policy Information          46         12.0             0.014218   
Social Media Offer Announcement          33         11.0             0.006402   
SEO          Health Tips                 33          1.0             0.007594   
Email        Covid Awareness             38          0.0             0.053969   
             Health Tips                 16          0.0             0.008424   
TV           Covid Awareness             24          0.0             0.014329   
             Health Awareness            17          0.0             0.035849   
             Policy Information          11          0.0             0.061336   

                                   cost_per_click  customer_acqisition_cost  \
platform     campaign_type                                                    
Social Media Product Promotion  

In [48]:
# df_campaign_type['signup_rate'] = (df_campaign_type['num_signups']/df_campaign_type['impressions'])*100
df_campaign_type.sort_values(by="signup_rate", ascending=False)

cost  impressions    clicks  num_signups  \
campaign_type                                                            
Health Awareness       4520.521931       820960  124169.0       3395.0   
Product Promotion     10965.805894      1562824  105050.5       5581.0   
Policy Information    10814.963725      1816321  265360.0       2987.0   
Customer Testimonial   8676.854990      1511676  109730.0       2229.0   
Health Tips            5154.659153       719101   50398.0        972.0   
Covid Awareness       13976.025031      1514805  125856.5       1064.0   
Offer Announcement     6081.417424      1134468   71704.0         61.0   

                      cost_per_impression  cost_per_click  \
campaign_type                                               
Health Awareness                 0.005506        0.036406   
Product Promotion                0.007017        0.104386   
Policy Information               0.005954        0.040756   
Customer Testimonial             0.005740        0.079075   
Health Tips                      0.007168        0.102279   
Covid Awareness                  0.009226        0.111047   
Offer Announcement               0.005361        0.084813   

                      customer_acqisition_cost  click_through_rate  \
campaign_type                                                        
Health Awareness                      1.331523           15.124854   
Product Promotion                     1.964846            6.721838   
Policy Information                    3.620678           14.609752   
Customer Testimonial                  3.892712            7.258831   
Health Tips                           5.303147            7.008473   
Covid Awareness                      13.135362            8.308429   
Offer Announcement                   99.695368            6.320496   

                      signup_rate  
campaign_type                      
Health Awareness         0.413540  
Product Promotion        0.357110  
Policy Information       0.164453  
Customer Testimonial     0.147452  
Health Tips              0.135169  
Covid Awareness          0.070240  
Offer Announcement       0.005377

In [49]:
df_platform= df.groupby(['platform'])[columns].sum().sort_values(by="num_signups", ascending=False)
df_platform = calculate_campign_metrics(df_platform)
df_platform.reset_index()

platform          cost  impressions    clicks  num_signups  \
0  Social Media  17089.572975      3259312  280847.5       7610.0   
1         Email  16678.967588      2022478  337928.5       4130.0   
2           SEO  21246.992710      3188926  233492.0       4055.0   
3            TV   5174.714875       609439       0.0        494.0   

   cost_per_impression  cost_per_click  customer_acqisition_cost  \
0             0.005243        0.060850                  2.245673   
1             0.008247        0.049356                  4.038491   
2             0.006663        0.090997                  5.239702   
3             0.008491             inf                 10.475131   

   click_through_rate  signup_rate  
0            8.616772     0.233485  
1           16.708637     0.204205  
2            7.321964     0.127159  
3            0.000000     0.081058

#### As we can see in the above table, the highest signup rate is through social media platform with 0.23%, followed by Email with 0.20%. Although, signup rate is high in social media campaigning, CTR is very high for campaining through Email, also CPC is lowest of all with 0.049$. Hence to increase brand awareness we should consider reallocation of budget for platforms, we can also observe that signup rate is very low for campaining through TV and CAC is very high with 10.47$.

In [50]:
# df_platform.reset_index()
fig =px.pie(
    data_frame=df_platform.reset_index(),
    names="platform",
    values="click_through_rate",
    title="CTR thorugh different platforms"
    )
fig.show()
fig.write_html("C:/Users/karthikeya/health_tech_analytics/plots/CTR thorugh different platforms.html")
fig.write_image("C:/Users/karthikeya/health_tech_analytics/plots/CTR thorugh different platforms.jpeg")

In [51]:
df_campaign_plat_reset = df.groupby(['platform','campaign_type'])[columns].sum()
df_campaign_plat_reset =df_campaign_plat_reset.reset_index()
df_campaign_plat_reset =calculate_campign_metrics(df_campaign_plat_reset)
df_campaign_plat_reset[df_campaign_plat_reset['platform']=="Email"]

platform         campaign_type         cost  impressions    clicks  \
0    Email       Covid Awareness  3171.410822        58764    8527.0   
1    Email  Customer Testimonial  1624.442464       428100   47330.0   
2    Email      Health Awareness  3063.671400       310540   89959.0   
3    Email           Health Tips   359.111694        42630    2316.5   
4    Email    Policy Information  5113.225610      1068155  175084.0   
5    Email     Product Promotion  3347.105598       114289   14712.0   

   num_signups  cost_per_impression  cost_per_click  customer_acqisition_cost  \
0          0.0             0.053969        0.371926                       inf   
1        586.0             0.003795        0.034322                  2.772086   
2       3281.0             0.009866        0.034056                  0.933761   
3          0.0             0.008424        0.155023                       inf   
4        230.0             0.004787        0.029204                 22.231416   
5         33.0             0.029286        0.227509                101.427442   

   click_through_rate  signup_rate  
0           14.510585     0.000000  
1           11.055828     0.136884  
2           28.968571     1.056547  
3            5.433967     0.000000  
4           16.391254     0.021532  
5           12.872630     0.028874

In [64]:
fig = px.bar(data_frame=df_campaign_plat_reset[df_campaign_plat_reset['platform']=="Email"],
             x= 'campaign_type',
             y='signup_rate',
             title="Email campaign performance by campaign type with signup rate")

fig.show()
fig.write_image(r'../plots/Email campaign performance by campaign type with signup rate.png')

In [53]:
fig = px.bar(data_frame=df_campaign_plat_reset[df_campaign_plat_reset['platform']=="Email"],
             x= 'campaign_type',
             y='click_through_rate',
             title="Email campaign performance by campaign type with CTR")

fig.show()
fig.write_image(r'../plots/Email campaign performance by campaign type with CTR.png')

In [68]:
import plotly.graph_objects as go

# Filter the data for Email platform
df_email = df_campaign_plat_reset[df_campaign_plat_reset['platform'] == "Email"]
df_sm = df_campaign_plat_reset[df_campaign_plat_reset['platform'] == "Social Media"]

# Create a bar chart with signup_rate
fig = go.Figure(data=[
    go.Bar(name='Email', x=df_email['campaign_type'], y=df_email['signup_rate']),
    go.Bar(name='Social Media', x=df_sm['campaign_type'], y=df_sm['signup_rate'])
])

# Update the layout
fig.update_layout(
    title='EEmail and Social Media channels campaign performance by campaign type combined',
    xaxis_title='Campaign Type',
    yaxis_title='Signup Rate',
    barmode='group'
)

fig.show()
fig.write_image(r'../plots/Email and Social Media channels campaign performance by campaign type combined.png')

### As we saw in the previous table, highest CTR is for campaigns through Email, further if we cateogrise platform type and campagin type, with previous knowledge in mind, in Email campaigns Health Awareness campaign is very effective with signup rate of 1.05% also has the lowest CAC with 0.933$ and highest CTR with 28.96%. Covid awareness and Health Tips campagins seems to be ineffctive and costing us about ~3550$ approx. Hence it is wise to consider the allocation of budgets in Email campaigning keeping in mind this suggestion is for customer acquisition.  For covid awareness campaign, CTR is next to the Policy Information campaign with 14.51% in the third position.

In [54]:
fig = px.pie(
    data_frame=df_campaign_plat_reset[df_campaign_plat_reset['platform']=="Email"],
    values="click_through_rate",
    names="campaign_type",
    title=" CTR for different types of campaigns through Email"
)
fig.show()
fig.write_html(r'C:\Users\karthikeya\health_tech_analytics\plots\CTR for different types of campaigns through Email.html')
fig.write_image(r'C:\Users\karthikeya\health_tech_analytics\plots\CTR for different types of campaigns through Email.png')

In [55]:
df_campaign_plat_reset[df_campaign_plat_reset['platform']=="SEO"].sort_values(by="signup_rate",ascending=False)

platform         campaign_type         cost  impressions   clicks  \
12      SEO     Product Promotion  2654.454476       116890  25243.5   
7       SEO  Customer Testimonial  4389.367436       589766  44160.0   
6       SEO       Covid Awareness  6798.282804       914491  54274.5   
8       SEO      Health Awareness   394.641083       480790  34210.0   
11      SEO    Policy Information  1912.411772       134509  32081.0   
10      SEO    Offer Announcement  3942.507148       800350  29349.0   
9       SEO           Health Tips  1155.327990       152130  14174.0   

    num_signups  cost_per_impression  cost_per_click  \
12       1273.0             0.022709        0.105154   
7        1588.0             0.007443        0.099397   
6        1017.0             0.007434        0.125257   
8         114.0             0.000821        0.011536   
11         12.0             0.014218        0.059612   
10         50.0             0.004926        0.134332   
9           1.0             0.007594        0.081510   

    customer_acqisition_cost  click_through_rate  signup_rate  
12                  2.085196           21.595945     1.089058  
7                   2.764085            7.487715     0.269259  
6                   6.684644            5.934941     0.111209  
8                   3.461764            7.115373     0.023711  
11                159.367648           23.850449     0.008921  
10                 78.850143            3.667021     0.006247  
9                1155.327990            9.317031     0.000657

In [56]:
fig = px.pie(
    data_frame=df_campaign_plat_reset[df_campaign_plat_reset['platform']=="SEO"].sort_values(by="click_through_rate",ascending=False),
    values="click_through_rate",
    names="campaign_type",
    title=" CTR for different types of campaigns through SEO"
)
fig.show()
fig.write_html(r'C:\Users\karthikeya\health_tech_analytics\plots\CTR for different types of campaigns through SEO.html')
fig.write_image(r'C:\Users\karthikeya\health_tech_analytics\plots\CTR for different types of campaigns through SEO.png')

In [58]:
df_campaign_plat_reset = df.groupby(['platform','campaign_type'])[columns].sum()
df_campaign_plat_reset = df_campaign_plat_reset.reset_index(drop=False)
df_campaign_plat_reset = calculate_campign_metrics(df_campaign_plat_reset)
df_campaign_plat_reset[df_campaign_plat_reset['platform']=="Social Media"].sort_values(by="click_through_rate", ascending=False)

platform         campaign_type         cost  impressions   clicks  \
13  Social Media       Covid Awareness  3133.144076       480610  63055.0   
16  Social Media    Offer Announcement  2138.910275       334118  42355.0   
17  Social Media    Policy Information  2535.793508       593220  58195.0   
18  Social Media     Product Promotion  2978.460558       833213  65095.0   
15  Social Media           Health Tips  3640.219468       524341  33907.5   
14  Social Media  Customer Testimonial  2663.045090       493810  18240.0   

    num_signups  cost_per_impression  cost_per_click  \
13         47.0             0.006519        0.049689   
16         11.0             0.006402        0.050500   
17       2745.0             0.004275        0.043574   
18       3781.0             0.003575        0.045756   
15        971.0             0.006942        0.107357   
14         55.0             0.005393        0.146000   

    customer_acqisition_cost  click_through_rate  signup_rate  
13                 66.662640           13.119785     0.009779  
16                194.446389           12.676659     0.003292  
17                  0.923786            9.810020     0.462729  
18                  0.787744            7.812528     0.453786  
15                  3.748939            6.466689     0.185185  
14                 48.419002            3.693728     0.011138

In [62]:
fig = px.bar(data_frame=df_campaign_plat_reset[df_campaign_plat_reset['platform']=="Social Media"].sort_values(by="click_through_rate", ascending=False),
             x='campaign_type',
             y='signup_rate',
             title="Social Media campaign performance by campaign type with signup rate")

fig.show()
fig.write_image(r'../plots/social media campaign performance by campaign type with signup rate.png')

In [60]:
fig = px.bar(data_frame=df_campaign_plat_reset[df_campaign_plat_reset['platform']=="Social Media"].sort_values(by="click_through_rate", ascending=False),
             x='campaign_type',
             y='click_through_rate',
             title="Social Media campaign performance by campaign type with CTR")

fig.show()
fig.write_image(r'../plots/social media campaign performance by campaign type with CTR.png')

### Covid Awareness campaign through social media paltform does seem to do very well in terms of CTR with 13.119%, but if we look at the number of signups, the number is very negligible 47 compared to product promotion with 3781. But in terms of brand awareness Covid Awareness and Offer Announcement seems to do well with 13.11% and 12.67%. In terms of signups, Product promotion is ahead with 0.46% and 3781 signups, also CAC is 0.78$, followed by Policy Information with 2745 signups and 0.92$ CAC. 

In [ ]:
fig = px.pie(
    data_frame=df_campaign_plat_reset[df_campaign_plat_reset['platform']=="Social Media"].sort_values(by="click_through_rate",ascending=False),
    values="click_through_rate",
    names="campaign_type",
    title=" CTR for different types of campaigns through Social Media"
)
fig.show()
fig.write_html(r'C:\Users\karthikeya\health_tech_analytics\plots\CTR for different types of campaigns through Social Media.html')
fig.write_image(r'C:\Users\karthikeya\health_tech_analytics\plots\CTR for different types of campaigns through Social Media.png')

: 

### To acquire customers we need to invest in campaigns in which signups is high. In this, we can see, in social media platform, product promotion campagin type have the highest signups,

In [ ]:
df_platform['signup_rate'] = df_platform['num_signups']/df_platform['impressions']*100
df_platform.sort_values(by=["num_signups","customer_acquisition_cost"],ascending=False)

: 

### Just by counting the number of signups coming from TV, we need to evaluate more than this while allocating the budget to TV platfor. But if we have to go with the numbers present here, it is recommended to stop TV campaigns as we spend not so high on TV but even that is not contributing much to signup rates

In [ ]:
fig = px.bar(data_frame=df_platform.reset_index(),
             x="platform",
             color="cost",
             y ="num_signups",
             title="Number of signups across different platforms"
)
fig.show()
fig.write_html(r'C:\Users\karthikeya\health_tech_analytics\plots\signups across different platforms.html')
fig.write_image(r'C:\Users\karthikeya\health_tech_analytics\plots\signups across different platforms.png')

: 

: 